packages/mui-lab/src/useSnackbars/useSnackbars.ts

```diff
@@ -0,0 +1,12 @@
+import * as React from 'react';
+import SnackbarsContext, { SnackbarsContextProps } from '../SnackbarsProvider/SnackbarsContext';
+
+export default function useSnackbars(): SnackbarsContextProps {
+  const context = React.useContext(SnackbarsContext);
+
+  if (context === undefined) {
+    throw new Error('MUI: useSnackbars must be within a SnackbarsProvider');
```

Comment:
```suggestion
    throw new Error('MUI: useSnackbars must be within a SnackbarsProvider/');
```

I think that we could improve this error message. We could differentiate what went wrong to how to fix the problem, and for fixing the problem, maybe even link to the docs with a https://mui.com/r/y link.

I also wonder if the lab shouldn't use `MuiError`.

Reply:
Improved error message in https://github.com/mui/material-ui/pull/31991/commits/834bf996837777ae65f99948c41b35f9d0286e5b and https://github.com/mui/material-ui/pull/31991/commits/4ba165f2da2d108dfbdfba2bc16411da4cd2c353